libs/coin-framework/src/utils.ts

```diff
@@ -0,0 +1,11 @@
+import BigNumber from "bignumber.js";
+
+export function fromBigNumberToBigInt<T>(
+  bigNumber: BigNumber | undefined,
+  defaultValue?: T,
+): bigint | T {
+  if (bigNumber != null) {
+    return BigInt(bigNumber.toString());
+  }
+  return defaultValue as T;
+}
```

Comment:
unsure if that's idiomatic TS ?

Looks like this function serves 2 purposes:
- either converting BigNumber to bigint
- either casting nulls to T

It feels strange for a scala dev

Reply:
I wanted the method to be able to set null / undefined as default value

We could just force defaultValue to be a BigInt or BigNumber if you prefer

Reply:
no need